webapp/dojo_news/src/App.spec.js

```
@@ -0,0 +1,221 @@
+import {shallowMount} from '@vue/test-utils';
+import App from "./App";
+import 'regenerator-runtime';
+import Vue from "vue";
+
+describe("News list", () => {
+    it('does not render any items on an initially empty list of NewsItems', () => {
+        const wrapper = shallowMount(App, {
+            propsData: {
+                initialNewsListItems: []
+            }
+        });
+        expect(wrapper.find("#newslist").html()).toBe('<div id="newslist"></div>');
+        expect(wrapper.findAll(".newslistitem").length).toBe(0);
+    });
+
+    it('renders items on an initially filled list of NewsItems', () => {
+        const wrapper = shallowMount(App, {
+            propsData: {
+                initialNewsListItems: [
+                    {id: 0, title: "macOS", votes: 0},
+                    {id: 1, title: "Linux", votes: 0},
+                    {id: 2, title: "Windows", votes: 0}
+                ]
+            }
+        });
+        expect(wrapper.find("#newslist").html()).not.toBe('<div id="newslist"></div>');
```

```suggestion
```

Testing nothingness is really error-prone and can lead to false negatives. If you have to, use
```
Arrange - Assert - Act - Assert
```
to avoid silently passing tests.

---

Switched to testing the element not to be an empty DOM element. Better?